### PR TITLE
Support keyword `SERIAL`

### PR DESCRIPTION
**Pull request by spongedu**
### What problem does this PR solve? 

for #467 , support keyword `SERIAL`.

### What is changed and how it works?

1. Support `SERIAL` as a `Numeric Type` in `ColumnDef`
2. Support `SERIAL DEFAULT VALUE` as an `ColumnOption`
3. Add related tests

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

N / A

Side effects

N / A

### CODE REVIEW

**Comment by spongedu** [comment]
it's interesting that the `Verify parser.go is up-to-date` stage in tests is not consistent with the `fmt` stage. The tests will fail anyway...

**Comment by spongedu** [comment]
I think we should add a `fmt` command in `Verify parser.go is up-to-date` stage in  the `ci` test base

**Comment by spongedu** [comment]
@leoppro  PTAL :)

File: parser_test.go [hunk 1]
```diff
@@ -2426,6 +2426,15 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"create table nchar (a int, b nchar);", true, "CREATE TABLE `nchar` (`a` INT,`b` CHAR)"},
 		{"create table nchar (a int, b nchar(50));", true, "CREATE TABLE `nchar` (`a` INT,`b` CHAR(50))"},
 
+		// Test keyword `SERIAL`
+		{"create table t (a serial);", true, "CREATE TABLE `t` (`a` BIGINT UNSIGNED NOT NULL AUTO_INCREMENT UNIQUE KEY)"},
+		{"create table t (a serial null);", true, "CREATE TABLE `t` (`a` BIGINT UNSIGNED NOT NULL AUTO_INCREMENT UNIQUE KEY NULL)"},
```
Comment (leoppro) [inline parser_test.go]:
```suggestion
		{"create table t (a serial null);", true, "CREATE TABLE `t` (`a` BIGINT UNSIGNED AUTO_INCREMENT UNIQUE KEY)"},
```
this should be right

File: parser_test.go [hunk 2]
```diff
@@ -2426,6 +2426,15 @@ func (s *testParserSuite) TestDDL(c *C) {
...
+		{"create table t (b int, a serial);", true, "CREATE TABLE `t` (`b` INT,`a` BIGINT UNSIGNED NOT NULL AUTO_INCREMENT UNIQUE KEY)"},
+		{"create table t (a int serial default value);", true, "CREATE TABLE `t` (`a` INT NOT NULL AUTO_INCREMENT UNIQUE KEY)"},
+		{"create table t (a int serial default value null);", true, "CREATE TABLE `t` (`a` INT NOT NULL AUTO_INCREMENT UNIQUE KEY NULL)"},
```
Comment (leoppro) [inline parser_test.go]:
```suggestion
		{"create table t (a int serial default value null);", true, "CREATE TABLE `t` (`a` INT AUTO_INCREMENT UNIQUE KEY)"},
```

File: parser_test.go (hunk 1 above)
Comment (spongedu) [reply parser_test.go]: The `Restore` method only require the restored sql is semantic equivalence to the origin, so  
I just make sure the semantic equivalence here,  leave the duplicate removal job behind.

The current codebase fail on this kind of sql as well (`create table t (a serial not null null not null null not null);` for example), Shall I fix this in this pr?

File: parser_test.go (hunk 2 above)
Comment (spongedu) [reply parser_test.go]: ditto

File: parser_test.go (hunk 1 above)
Comment (tangenta) [reply parser_test.go]: I think it's TiDB's responsibility to remove duplication. @leoppro 
If we do this completely in the parser, tons of code in TiDB needs to be move.

Comment (leoppro) [reply parser_test.go]: @kennytm what do you think about it

Comment (kennytm) [reply parser_test.go]: Let's leave it as-is.

**Comment by leoppro** [comment]
LGTM please resolve conflicts
@spongedu

**Pull request merged**